drizzlepac/haputils/catalog_utils.py

```diff
@@ -2777,14 +2777,15 @@ def _evaluate_segmentation_image(self, segm_img, image_data, big_island_only=Fal
         nbins = segm_img.max_label
         log.info("Number of sources from segmentation map: %d", nbins)
 
-        n, binedges = np.histogram(segm_img.data, range=(1, nbins))
-        real_pixels = (image_data != 0).sum()
-        biggest_source = n.max()/float(real_pixels)
-        log.info("Biggest_source: %f", biggest_source)
+        # Compute the biggest source identified 
+        real_pixels = (np.isfinite(image_data) & (image_data != 0)).sum().clip(min=1)
+        biggest_source = segm_img.areas.max()/real_pixels
```

Comment:
It's cool that this `areas` attribute already exists in the `SegmentationImage` object.  This is definitely better than computing it (as I had suggested).

Reply:
Very cool.